### PR TITLE
Bump the k8s dependencies to 1.22.1 and controller-runtime to 0.10.0

### PR DESCRIPTION
**Pull request by timflannagan**
Updates the controller runtime release to v0.10.0, which has been rebased on top of k8s v1.22.x dependencies.

Bumps the k8s.io dependencies to v1.22.x.

Builds were failing due to the openapi generated packages:

```bash
$ make build
 ./build/build.sh
building marketplace-operator...
# github.com/operator-framework/operator-marketplace/pkg/apis/operators/v2
pkg/apis/operators/v2/zz_generated.openapi.go:36:3: cannot use "github.com/go-openapi/spec".Schema{...} (type "github.com/go-openapi/spec".Schema) as type "k8s.io/kube-openapi/pkg/validation/spec".Schema in field value
pkg/apis/operators/v2/zz_generated.openapi.go:57:8: cannot use ref("k8s.io/apimachinery/pkg/apis/meta/v1.ObjectMeta") (type "k8s.io/kube-openapi/pkg/validation/spec".Ref) as type "github.com/go-openapi/spec".Ref in field value
pkg/apis/operators/v2/zz_generated.openapi.go:62:8: cannot use ref("github.com/operator-framework/operator-marketplace/pkg/apis/operators/v2.CatalogSourceConfigSpec") (type "k8s.io/kube-openapi/pkg/validation/spec".Ref) as type "github.com/go-openapi/spec".Ref in field value
pkg/apis/operators/v2/zz_generated.openapi.go:67:8: cannot use ref("github.com/operator-framework/operator-marketplace/pkg/apis/operators/v2.CatalogSourceConfigStatus") (type "k8s.io/kube-openapi/pkg/validation/spec".Ref) as type "github.com/go-openapi/spec".Ref in field value
# github.com/operator-framework/operator-marketplace/pkg/apis/operators/v1
pkg/apis/operators/v1/zz_generated.openapi.go:36:3: cannot use "github.com/go-openapi/spec".Schema{...} (type "github.com/go-openapi/spec".Schema) as type "k8s.io/kube-openapi/pkg/validation/spec".Schema in field value
pkg/apis/operators/v1/zz_generated.openapi.go:57:8: cannot use ref("k8s.io/apimachinery/pkg/apis/meta/v1.ObjectMeta") (type "k8s.io/kube-openapi/pkg/validation/spec".Ref) as type "github.com/go-openapi/spec".Ref in field value
pkg/apis/operators/v1/zz_generated.openapi.go:62:8: cannot use ref("github.com/operator-framework/operator-marketplace/pkg/apis/operators/v1.OperatorSourceSpec") (type "k8s.io/kube-openapi/pkg/validation/spec".Ref) as type "github.com/go-openapi/spec".Ref in field value
pkg/apis/operators/v1/zz_generated.openapi.go:67:8: cannot use ref("github.com/operator-framework/operator-marketplace/pkg/apis/operators/v1.OperatorSourceStatus") (type "k8s.io/kube-openapi/pkg/validation/spec".Ref) as type "github.com/go-openapi/spec".Ref in field value
make: *** [Makefile:23: osbs-build] Error 2
```

Towards the later end of the 4.9 release, work was done to remove the operator-sdk dependencies, which was previously responsible for generating these schema validation files through the `operator-sdk generate ...` command(s). Now, we'll need to directly run this command ourselves.

Introduces a tools.go file that adds the k8s.io/kube-openapi as a build dependency. A new helper bash script, hack/openapi-gen.sh, has been introduced as well, which allows us to now run that build dependency directory, generating the v1 and v2 schema validation files. The Makefile has been updated to add the `generate` target, which runs that script, and the `manifests` target has been updated to add the `generate` target as a prerequisite.

### CODE REVIEW

**Comment by timflannagan** [comment]
Note: it seems like we can remove the v1 and v2 marketplace APIs entirely as they've been properly deprecated/removed downstream. I'd like to tackle doing something like that outside of this PR as there's some complexity and decoupling of those types in the codebase that's still necessary.

File: go.mod [hunk 1]
```diff
@@ -6,13 +6,11 @@ require (
 	cloud.google.com/go v0.81.0 // indirect
 	github.com/emicklei/go-restful v2.15.0+incompatible // indirect
 	github.com/fsnotify/fsnotify v1.4.9
-	github.com/go-openapi/spec v0.19.5
```
Comment (timflannagan) [inline go.mod:9]: Note: it looks like openapi generation was deprecated in v0.17.x: https://sdk.operatorframework.io/docs/upgrading-sdk-version/version-upgrade-guide/#v017x.

**Review by anik120 — APPROVED** [review]
/lgtm

**Comment by timflannagan** [comment]
That test needs to be updated.

/hold

**Comment by timflannagan** [comment]
Test should be fixed now :crossed_fingers: 

/retest
/hold cancel

**Comment by timflannagan** [comment]
Manually re-adding the lgtm label - I added a commit to continue reducing the number of testing flakes.

**Pull request merged**